tests/integration/suite/actors/rebalancing-double-activation/rebalancing-double-activation.go

```diff
@@ -0,0 +1,416 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or impliei.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package healthz
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	chi "github.com/go-chi/chi/v5"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	grpcinsecure "google.golang.org/grpc/credentials/insecure"
+
+	placementv1pb "github.com/dapr/dapr/pkg/proto/placement/v1"
+	"github.com/dapr/dapr/tests/integration/framework"
+	"github.com/dapr/dapr/tests/integration/framework/process/daprd"
+	prochttp "github.com/dapr/dapr/tests/integration/framework/process/http"
+	"github.com/dapr/dapr/tests/integration/framework/process/placement"
+	"github.com/dapr/dapr/tests/integration/framework/util"
+	"github.com/dapr/dapr/tests/integration/suite"
+)
+
+func init() {
+	suite.Register(new(rebalancingDoubleActivation))
+}
+
+// Number of iterations for the test
+const iterations = 30
+
+// rebalancingDoubleActivation tests that during rebalancing, reminders do not cause actors to be activated on 2 separate hosts.
+type rebalancingDoubleActivation struct {
+	daprd              [2]*daprd.Daprd
+	srv                [2]*prochttp.HTTP
+	handler            [2]*httpServer
+	place              *placement.Placement
+	placementStream    placementv1pb.Placement_ReportDaprStatusClient
+	activeActors       []atomic.Bool
+	doubleActivationCh chan string
+}
+
+func (i *rebalancingDoubleActivation) Setup(t *testing.T) []framework.Option {
+	i.activeActors = make([]atomic.Bool, iterations)
+	i.doubleActivationCh = make(chan string)
+
+	// Remove the database if present
+	// Ignore errors here
+	_ = os.Remove("tmp/test-data.db")
```

Comment:
Mentioned above why `t.TmpDir` can't be used, because it needs to be available in multiple methods.

> why can't we use the in-memory state store here?

The in-memory state store is tied to a single sidecar. Here we need 2 sidecars to use the same actor store to be able to reproduce this bug.